### PR TITLE
$fullResult parameter for Elastica\Type::count()

### PR DESCRIPTION
**Pull request by vinusebastian**
Addresses https://github.com/ruflin/Elastica/issues/719

### CODE REVIEW

File: lib/Elastica/Search.php [hunk 1]
```diff
@@ -476,12 +476,11 @@ public function count($query = '', $fullResult = false)
 
         $query = $this->getQuery();
         $path = $this->getPath();
-
         $response = $this->getClient()->request(
             $path,
             Request::GET,
             $query->toArray(),
-            array(self::OPTION_SEARCH_TYPE => self::OPTION_SEARCH_TYPE_COUNT)
+            array(self::OPTION_SEARCH_TYPE => self::OPTION_SEARCH_TYPE_QUERY_THEN_FETCH)
```
Comment (ruflin) [inline lib/Elastica/Search.php]: Shouldn't this be dependent on which param is passed to count?

**Comment by ruflin** [comment]
@vinusebastian Thanks for the PR. Looks good to me. Just one small question above. Can you also update the CHANGELOG.md?

**Comment by ewgRa** [comment]
maybe better to create countResult?

also is OPTION_SEARCH_TYPE_QUERY_THEN_FETCH change not a break BC?

File: lib/Elastica/Search.php (hunk 1 above)
Comment (vinusebastian) [inline lib/Elastica/Search.php]: Yes, but Elasticsearch documentation says that `search type` of `count` is deprecated?  And that `count` doesn't have any advantage over `query then fetch` (provided size of query is 0)

https://www.elastic.co/guide/en/elasticsearch/reference/2.x/search-request-search-type.html

Comment (ruflin) [inline lib/Elastica/Search.php]: Wasn't aware of that. LGTM.

**Comment by vinusebastian** [comment]
@ewgRa Didn't follow what you said. Can you explain?

**Comment by ruflin** [comment]
I think this PR also overlaps with: https://github.com/ruflin/Elastica/pull/1038

File: lib/Elastica/Type.php [hunk 2]
```diff
@@ -358,17 +358,18 @@ public function search($query = '', $options = null)
...
-     * @return int number of documents matching the query
+     * @param string|array|\Elastica\Query $query       Array with all query data inside or a Elastica\Query object
+     * @param ResultSet                    $fullResult (default = false) By default only the total hit count is returned. If set to true, the full ResultSet including aggregations is returned.
+     * 
+     * @return int number of documents matching the query|ResultSet
      *
      * @see \Elastica\SearchableInterface::count
      */
-    public function count($query = '')
+    public function count($query = '', $fullResult = false)
```
Comment (ruflin) [inline lib/Elastica/Type.php]: It looks like the approach here with an options array could be more flexible: https://github.com/ruflin/Elastica/pull/1038/files#diff-1d7207a5d6e8ecfaa5d6386b7e0edb99R368 Can you have a look at this together with @intech ?

**Comment by ewgRa** [comment]
@vinusebastian never mind, seems I was wrong

File: lib/Elastica/Type.php (hunk 2 above)
Comment (vinusebastian) [inline lib/Elastica/Type.php]: Fair enough. Closing the PR

Comment (ruflin) [inline lib/Elastica/Type.php]: @vinusebastian Idea was not necessarly to close this one, but have a chat with @intech on the best approach :-)

**Pull request closed without merging**